quinn/examples/https_client.rs

```diff
@@ -52,28 +58,51 @@ fn main() {
             0
         }
     };
+
     ::std::process::exit(code);
 }
 
 #[tokio::main]
 async fn run(options: Opt) -> Result<()> {
-    let url = options.url;
-    let remote = (url.host_str().unwrap(), url.port().unwrap_or(4433))
-        .to_socket_addrs()?
-        .next()
-        .ok_or_else(|| anyhow!("couldn't resolve to an address"))?;
+    let url = options.url.clone();
+    let mut endpoint = initialize_endpoint(&options)?;
 
-    let mut endpoint = quinn::Endpoint::builder();
+    let quinn::NewConnection {
+        connection: mut conn, ..
+    } = initialize_connection(&options, &endpoint, url.clone()).await?;
+
+    if options.rebind {
+        let socket = std::net::UdpSocket::bind("[::]:0").unwrap();
+        let addr = socket.local_addr().unwrap();
+        eprintln!("rebinding to {}", addr);
+        endpoint.rebind(socket).expect("rebind failed");
+    }
+
+    // Perform a GET request with the filename.
+    perform_request(url, &mut conn).await;
+
+    // Close connection.
+    conn.close(0u32.into(), b"done");
+
+    Ok(())
+}
+
+/// Initializes the client configuration from the command-line arguments.
+fn initialize_client_configuration(options: &Opt) -> Result<quinn::ClientConfig> {
```

Comment:
Naming nit:
```suggestion
fn make_client_configuration(options: &Opt) -> Result<quinn::ClientConfig> {
```
and similar for the others. IMO this is more readable.